### PR TITLE
add `ControllerClient`

### PR DESCRIPTION
**Pull request by edgarriba**

### CODE REVIEW

File: py/farm_ng/controller/controller_client.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+from dataclasses import dataclass
+
+import grpc
+from farm_ng.controller import controller_pb2
+from farm_ng.controller import controller_pb2_grpc
+
+logging.basicConfig(level=logging.DEBUG)
+
+
+class ControllerServiceState:
```
Comment (ethanrublee) [inline py/farm_ng/controller/controller_client.py:12]: Suggest getting rid of this wrapper @edgarriba

File: py/farm_ng/controller/controller_client.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                self.logger.info("Got Cancelled Error")
+                break
+
+    async def get_state(self) -> ControllerServiceState:
+        state: ControllerServiceState
+        try:
+            response: controller_pb2.GetServiceStateResult = await self.stub.getServiceState(
+                controller_pb2.GetServiceStateRequest()
+            )
+            state = ControllerServiceState(response.state)
```
Comment (ethanrublee) [inline py/farm_ng/controller/controller_client.py:77]: just save the response.state protobuf in the class.

File: protos/farm_ng/controller/controller.proto [hunk 3]
```diff
@@ -80,7 +80,7 @@ message GetServiceStateRequest {
 }
 message GetServiceStateResult {
     string state_name = 1;
-    ServiceState state = 2;
+    ControllerServiceState state = 2;
```
Comment (Hackerman342) [inline protos/farm_ng/controller/controller.proto:83]: I still think we should standardize these generic proto definitions and have a common source for them. I don't see a reason for duplicating as `ControllerServiceState`, `CanbusServiceState`, `OakServiceState`, etc. when they could all just be `ServiceState`. But I guess we can deal with that later.

File: py/farm_ng/controller/controller_client.py (hunk 1 above)
Comment (edgarriba) [reply py/farm_ng/controller/controller_client.py:12]: the main motivation was purely debugging purposes. When you print protobuf enums you just get the value which makes difficult to understand without having the proto definition in front of you

File: protos/farm_ng/controller/controller.proto (hunk 3 above)
Comment (edgarriba) [reply protos/farm_ng/controller/controller.proto:83]: sure -- i believe we can re-factor code after we have in place all the implementation of the controller and state and make a first release

**Pull request merged**